yt_dlp/networking/__init__.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, List
+
+from ._urllib import UrllibRH  # noqa: F401
+from .common import Request, RequestHandler, Response
+from .exceptions import NoSupportingHandlers, RequestError, UnsupportedRequest
+from ..utils import bug_reports_message
+
+
+class RequestDirector:
+    """RequestDirector class
+
+    Helper class that, when given a request, forward it to a RequestHandler that supports it.
+
+    @param logger: Logger instance.
+    @param verbose: Print debug request information to stdout.
+    """
+
+    def __init__(self, logger=None, verbose=False):
```

Comment:
Since this was building on logger refactor it should have the default logger here. Sonce we dont have that yet, o think adding a `TODO(Grub4K): default logger` and making it required should suffice, so we can merge logger later